src/main/java/mate/academy/lib/BookDaoImpl.java

```diff
@@ -0,0 +1,105 @@
+package mate.academy.lib;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String URL = "jdbc:mysql://localhost:3306/book_schema?serverTimezone=UTC";
+    private static final String USER = "your_username";
+    private static final String PASSWORD = "your_password";
```

Comment:
Avoid hardcoding database connection details such as URL, username, and password directly in the code. Consider using a configuration file or environment variables to manage these sensitive details securely.